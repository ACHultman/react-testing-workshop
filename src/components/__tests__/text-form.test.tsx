import { cleanup, fireEvent, render } from "@testing-library/react";
import TextForm from "../text-form";

describe("ListItem component", () => {
  afterEach(() => {
    cleanup();
  });

  it("matches snapshot", () => {
    const { baseElement } = render(
      <TextForm handleSubmit={jest.fn()} handleChange={jest.fn()} text={""} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("calls handleChange function when input changes", () => {
    const handleChange = jest.fn();
    const { getByTestId } = render(
      <TextForm
        handleSubmit={jest.fn()}
        handleChange={handleChange}
        text={""}
      />
    );

    const input: HTMLInputElement = getByTestId(
      "text-input"
    ) as HTMLInputElement;
    fireEvent.change(input, { target: { value: "anything..." } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it("calls handleSubmit function when submitting", () => {
    const handleSubmit = jest.fn();
    const { getByTestId } = render(
      <TextForm
        handleSubmit={handleSubmit}
        handleChange={jest.fn()}
        text={""}
      />
    );

    fireEvent.submit(getByTestId("form"));
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
