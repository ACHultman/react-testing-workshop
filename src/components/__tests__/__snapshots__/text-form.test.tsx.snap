// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component matches snapshot 1`] = `
<body>
  <div>
    <div>
      <form
        data-testid="form"
      >
        <label>
          <div>
            <input
              data-testid="text-input"
              type="text"
              value=""
            />
          </div>
        </label>
        <input
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>
</body>
`;
